### PR TITLE
Issue 315 - Add Chart Clamp To Settings Dropdown

### PR DESCRIPTION
**Pull request by fletchertyler914**
#315 Fix. Add Toggle In Settings Dropdown.

### CODE REVIEW

File: web/app/components/Exchange/PriceChartD3.jsx [hunk 1]
```diff
@@ -36,6 +36,7 @@ class CandleStickChartWithZoomPan extends React.Component {
         this.state = {
             enableTrendLine: false,
             enableFib: false,
+            clampEnabled: true,
```
Comment (svk31) [inline web/app/components/Exchange/PriceChartD3.jsx]: This variable isn't used anywhere

**Comment by fletchertyler914** [comment]
@svk31 the chart behavior you described exists already due to the way its rendering the chart on update. You can see this by checking out the bitshares/master branch without this update and repeating the steps you mentioned. Since this is outside of the scope of this feature, we can create a new bug to address the fix. I don't like it either, but I'm not sure there is a way around this with the current library.

File: web/app/components/Exchange/PriceChartD3.jsx (hunk 1 above)
Comment (fletchertyler914) [reply web/app/components/Exchange/PriceChartD3.jsx]: Removed and recommitted.

**Comment by svk31** [comment]
No I'm afraid this behaviour does not exist in the current version in staging without these changes.

**Comment by wmbutler** [comment]
@fletchertyler914 Should I push this to next Sprint? We are releasing tomorrow.

**Comment by fletchertyler914** [comment]
@svk31 I just pulled the current staging branch and it is doing this same behavior for me. If i zoom in on the chart and click the settings cog it will reset the zoom. Actually I just checked master, bitshares, and staging and they are all doing the same thing for me.  Is this not happening for anyone else?

[Here's a screencast](https://vimeo.com/233696842)

**Comment by fletchertyler914** [comment]
@svk31 I just cloned this repo, ran the master branch, and the behavior still exist. If it matters, I am running Chrome v60 on a Windows 10 machine. It's definitely an issue that was already there since this wasn't running my fork.

**Comment by wmbutler** [comment]
@svk31 This is a really weird disconnect here. Did you see tyler's screencast?

**Comment by svk31** [comment]
Ok I see what's happening, I have made a local modification of my react-stockcharts files as shown in this PR https://github.com/rrag/react-stockcharts/pull/332. That fixes the issue in the current version, but it reappears when the clamp is introduced.

**Pull request merged**